code/modules/mob/living/life.dm

```diff
@@ -160,15 +160,16 @@
 		setEarDamage(-1, max(ear_deaf, 1))
 
 /mob/living/proc/update_sight()
-	set_sight(0)
-	if(stat == DEAD || (eyeobj && !eyeobj.living_eye))
-		update_dead_sight()
-	else
-		update_living_sight()
+	if(!stop_sight_update)
```

Comment:
Flip this as a guard statement

if(stop_sight_update)
	return

[everything else]

Reply:
Change implemented.